src/com/google/javascript/jscomp/CommandLineRunner.java

```diff
@@ -1156,6 +1160,15 @@ protected void addWhitelistWarningsGuard(
   }
 
   @Override
+  protected void checkModuleName(String name)
+      throws FlagUsageException {
+    if (!TokenStream.isJSIdentifier(
+        extraModuleNameChars.matcher(name).replaceAll("_"))) {
```

Comment:
The easiest way to test this would be to utilize the `--json_streams` flag. So we'll need to merge #1313 first to support that.